Ghidra: Capa Explorer UI Integration (#1786)
mandiant/capa: capa/ghidra/README.md

$ capa --version
```

3. Copy [capa_explorer.py](capa/ghidra/capa_explorer.py) and [capa_ghidra.py](/capa/ghidra/capa_ghidra.py) to your `$USER_HOME/ghidra_scripts` directory or manually add `</paths/to/scripts.py/>` to the Ghidra Script Manager.

## Usage

> After completing the installation steps you can execute `capa_explorer.py` using the Ghidra Script Manager to integrate capa's results with Ghidra's UI. `capa_ghidra.py` is designed to be ran with both the Ghidra Script Manager and Headless Analyzer for capa's normal reports.  

Reviewer: ```suggestion
After completing the installation steps you can execute `capa_explorer.py` and `capa_ghidra.py` using the Ghidra Script Manager. You can also execute `capa_ghidra.py` using Ghidra's Headless Analyzer.
```

### Ghidra Script Manager

To execute `capa_explorer.py` using the Ghidra Script Manager, first open the Ghidra Script Manager by navigating to `Window > Script Manager` in the Ghidra Code Browser. Next, locate `capa_explorer.py` by selecting the `Python 3 > capa` category or using the Ghidra Script Manager search funtionality. Finally, double-click `capa_explorer.py` to execute the script. If you don't see `capa_explorer.py`, make sure you have copied the script to your `$USER_HOME/ghidra_scripts` directory or manually added `</path/to/capa_explorer.py/>` to the Ghidra Script Manager.

> When executed, `capa_explorer.py` asks you to provide your capa rules directory. `capa_ghidra.py` may be ran in the same manner and supports `default`, `verbose`, and `vverbose` output formats when executed from the Ghidra Script Manager. `capa_ghidra.py` writes output to the Ghidra Console Window. `capa_explorer.py` will integrate results directly into the Ghidra project via labels, namespaces, comments, and bookmarks.

Reviewer: ```suggestion
Both scripts ask you to provide the path of your capa rules directory. `capa_ghidra.py` also asks you to select `default`, `verbose`, and `vverbose` output formats used when writing output to the Ghidra Console Window.
```

Viewing results in Ghidra Console Window:
<img src="/doc/img/ghidra_script_mngr_output.png">

> ### Interpreting `capa_explorer.py` Results

Reviewer: ```suggestion
### Using `capa_explorer.py`
```

> By invoking capa analysis via `capa_explorer.py`, this script will take the results and import them directly into the Ghidra project. In the Symbol Tree Window, under the Namespaces section, you can find the matched rules as well as the corresponding functions that contain the extracted features.

Reviewer: ```suggestion
`capa_explorer.py` integrates capa results directly with Ghidra. In the Symbol Tree Window, under the Namespaces section, you can find the matched rules as well as the corresponding functions that contain the matched features.
```

> Namespaces & Labeled Functions:

Reviewer: ```suggestion
Namespaces and labeled functions:
```

![image](https://github.com/mandiant/capa/assets/66766340/eeae33f4-99d4-42dc-a5e8-4c1b8c661492)

> Labeled functions may be clicked on from the Symbol Tree Window in order to hop straight to these sections of code in the Disassembly Listing and Decompilation views. Plate comments will be found above the function symbol, containing each matched rule for that function. Throughout the listing and decompilation views, you will also find pre-comments that highlight extracted features and the rules they correspond to.

Reviewer: ```suggestion
Labeled functions may be clicked in the Symbol Tree Window to navigate the Disassembly Listing and Decompilation windows to the function locations. A comment listing each matched capa rule is inserted at the beginning of the function and a comment for each matched capa feature is added at the matched address within the function. These comments can be viewed using both the Disassembly Listing and Decompilation windows.
```

> Commented Matches and Subscoped Features:

Reviewer: ```suggestion
Comments:
```

![image](https://github.com/mandiant/capa/assets/66766340/bb2b4170-7fd4-45fc-8c7b-ff8f2e2f101b)

> This script will also create Bookmarks on functions that may be categorized under the MITRE ATT&CK Framework as well as the Malware Behavior Catalog (MBC). These may be found via the Bookmarks Window, and the Labels may also be double clicked to align the Disassembly Listing and Decompilation views. 

Reviewer: ```suggestion
Bookmarks are added for capa matches that are categorized under the MITRE ATT&CK Framework and Malware Behavior Catalog (MBC). These may be found and navigated using Ghidra's Bookmarks Window.
```

> MITRE ATT&CK & MBC Bookmarks:

Reviewer: ```suggestion
MITRE ATT&CK & MBC bookmarks:
```
